Dodanie edytowania i usuwania zestawu komputerowego

oliwiabrozek opened the pull request.

oliwiabrozek committed:
Dodanie edytowania i usuwania zestawu komputerowego

bartlomiej-szlachta commented on backend/src/main/java/org/polsl/backend/service/ComputerSetService.java:
```diff
@@ -94,10 +94,48 @@ public void createComputerSet(ComputerSetInputDTO request) {
+    ComputerSet computerSet = computerSetRepository.findById(id)
+            .orElseThrow(() -> new NotFoundException("zestaw komputerowy", "id", id));
+    computerSet.setValidTo(LocalDateTime.now());
+    computerSetRepository.save(computerSet);
```
Przy usuwaniu na pewno trzeba jeszcze ustawić `valid_to` na bieżącą datę w obiektach klas pośredniczących

bartlomiej-szlachta commented:
Niestety, logika na edycję jest bardziej skomplikowana 😟 A mianowicie w momencie otrzymania żądania o edycję, która nie zmienia powiązań, nie powinniśmy dodawać nowego rekordu do bazy.

bartlomiej-szlachta commented:
I jeszcze w momencie zmiany powiązania z jakąś inną tabelą, w ostatnim wpisie z tabel pośredniczących trzeba ustawić `valid_to` na bieżącą datę

oliwiabrozek committed:
Merge branch 'master' into feature-cs-edit-del

# Conflicts:
#	backend/src/main/java/org/polsl/backend/controller/ComputerSetController.java

oliwiabrozek committed:
Uwzględnienie zmiany pola valid_to w tabelach pośredniczących podczas usuwania sprzętu komputerowego

oliwiabrozek committed:
Merge remote-tracking branch 'remotes/origin/master' into feature-cs-edit-del

# Conflicts:
#	backend/src/test/java/org/polsl/backend/SoftwareControllerIntegrationTest.java

+ EDYCJA ZESTAWU KOMPUTEROWEGO

bartlomiej-szlachta-softiq committed:
Dodanie informacji o błędach

bartlomiej-szlachta-softiq committed:
Aktualizacja komentarzy

oliwiabrozek committed:
Merge remote-tracking branch 'remotes/origin/master' into feature-cs-edit-del

# Conflicts:
#	backend/src/main/java/org/polsl/backend/controller/HardwareController.java
#	backend/src/main/java/org/polsl/backend/repository/AffiliationHardwareRepository.java
#	backend/src/main/java/org/polsl/backend/repository/ComputerSetHardwareRepository.java
#	backend/src/main/java/org/polsl/backend/service/HardwareService.java
#	backend/src/test/java/org/polsl/backend/HardwareControllerIntegrationTest.java

oliwiabrozek committed:
Dodana brakująca metoda do repozytorium ComputerSetHardware po mergu

oliwiabrozek committed:
Dodanie kilku zabezpieczeń

bartlomiej-szlachta-softiq committed:
hint :*

oliwiabrozek committed:
Zmiana sposobu sprawdzania
było ids-połączenia
jest ids-ids
+ ustawienie hardwareIds i softwareIds na @NotNull
(można przekazywać pustą listę ale nie można jej nie przekazać)

oliwiabrozek committed:
Merge remote-tracking branch 'origin/feature-cs-edit-del' into feature-cs-edit-del

# Conflicts:
#	backend/src/main/java/org/polsl/backend/service/ComputerSetService.java

bartlomiej-szlachta-softiq committed:
Zapisywanie nowych rekordów

bartlomiej-szlachta-softiq committed:
Poprawka - błąd znaleziony

bartlomiej-szlachta-softiq committed:
Poprawka - uwzględnienie tego, że może istnieć więcej niż jeden rekord dla tej samej pary w bazie pośredniczącej

bartlomiej-szlachta-softiq committed:
Poprawka przy usuwaniu

oliwiabrozek committed:
Merge remote-tracking branch 'remotes/origin/master' into feature-cs-edit-del

# Conflicts:
#	backend/src/main/java/org/polsl/backend/controller/SoftwareController.java
#	backend/src/main/java/org/polsl/backend/service/SoftwareService.java
#	backend/src/test/java/org/polsl/backend/SoftwareControllerIntegrationTest.java

oliwiabrozek committed:
Merge z masterem
